File: GeoBroker-Commons/src/main/java/de/hasenburg/geobroker/commons/model/message/ReasonCode.java
package de.hasenburg.geobroker.commons.model.message;

@SuppressWarnings("SpellCheckingInspection")
public enum ReasonCode {

	NormalDisconnection,
	ProtocolError,
	NotConnected,
	GrantedQoS0,
	Success,
	NoMatchingSubscribers,

	// New Reason Codes
	LocationUpdated,
	WrongBroker,
	NoMatchingSubscribersButForwarded //  locally there are no subscribers, but others MIGHT have some
}

File: GeoBroker-Commons/src/main/kotlin/de/hasenburg/geobroker/commons/model/message/ReasonCode.kt
package de.hasenburg.geobroker.commons.model.message

import kotlinx.serialization.Serializable

@Serializable
enum class ReasonCode {
    NormalDisconnection, //
    ProtocolError, //
    NotConnectedOrNoLocation, //
    GrantedQoS0, //
    Success, //
    NoMatchingSubscribers, //
    NoSubscriptionExisted, //

    // New Reason Codes
    LocationUpdated, //
}
